services/brig/src/Brig/SMTP.hs

```diff
@@ -50,17 +67,68 @@ deriveJSON defaultOptions {constructorTagModifier = map toLower} ''SMTPConnType
 
 makeLenses ''SMTP
 
-initSMTP :: Logger -> Text -> Maybe PortNumber -> Maybe (Username, Password) -> SMTPConnType -> IO SMTP
-initSMTP lg host port credentials connType = do
+data SMTPFailure = Unauthorized | ConnectionTimeout | CaughtException SomeException
+  deriving (Show)
+
+data SMTPPoolException = SMTPUnauthorized | SMTPConnectionTimeout
+  deriving (Eq, Show)
```

Comment:
why isn't one type enough?  maybe add a line of haddocks answering that question to future me, too?

Reply:
I was hesitant to throw wrapped exceptions (`CaughtException SomeException`) as I didn't want to change the original behavior regarding things that might be caught elsewhere... :thinking